Unit test broken

It seems our nightly builds have caught a problem with one test:

```Text
=========================== short test summary info ============================
FAILED niworkflows/tests/test_viz.py::test_cifti_surfaces_plot - ValueError: ...
===== 1 failed, 352 passed, 35 skipped, 104 warnings in 469.94s (0:07:49) ======
```

--- a/.circleci/config.yml
+++ b/.circleci/config.yml
@@ -104,7 +104,7 @@ jobs:
           name: Setup git-annex, DataLad & TemplateFlow
           command: |
             apk update && apk add --no-cache coreutils
-            conda install -y -c conda-forge datalad git wget
+            conda install -y -c conda-forge python=*=*cpython datalad git wget
             python -m pip install --no-cache-dir -U datalad-osf templateflow
             git config --global user.name 'NiPreps Bot'
             git config --global user.email 'nipreps@gmail.com'

--- a/.github/workflows/pythonpackage.yml
+++ b/.github/workflows/pythonpackage.yml
@@ -114,14 +114,14 @@ jobs:
     runs-on: ubuntu-latest
     strategy:
       matrix:
-        python-version: [3.7, 3.8, 3.9]
+        python-version: [3.7, 3.8, 3.9, "3.10"]
         install: [repo]
         include:
-          - python-version: 3.9
+          - python-version: 3
             install: sdist
-          - python-version: 3.9
+          - python-version: 3
             install: wheel
-          - python-version: 3.9
+          - python-version: 3
             install: editable
 
     env:
@@ -196,15 +196,9 @@ jobs:
     runs-on: ubuntu-latest
     strategy:
       matrix:
-        python-version: [3.7, 3.8, 3.9]
+        python-version: [3.7, 3.8, 3.9, "3.10"]
         install: [repo]
         pip-flags: ['--pre']
-        include:
-          # VTK won't build official 3.10 wheels before 9.2.0
-          # PyVista has posted dev versions, at least
-          - python-version: '3.10'
-            install: repo
-            pip-flags: '--pre --find-links https://wheels.pyvista.org/'
 
     env:
       INSTALL_TYPE: ${{ matrix.install }}

--- a/niworkflows/tests/generate_data.py
+++ b/niworkflows/tests/generate_data.py
@@ -39,6 +39,9 @@ def _create_dtseries_cifti(timepoints, models):
                 brain_structure=name,
             )
             setattr(bm, attr, indices)
+            if model_type == "CIFTI_MODEL_TYPE_SURFACE":
+                # define total vertices for surface models
+                setattr(bm, "surface_number_of_vertices", 32492)
             index_offset += len(data)
             brain_models.append(bm)
             timeseries = np.column_stack((timeseries, data.T))

--- a/niworkflows/viz/plots.py
+++ b/niworkflows/viz/plots.py
@@ -1035,7 +1035,8 @@ def cifti_surfaces_plot(
     # as potential nonsteady states
     data = img.dataobj[5:20].mean(axis=0)
 
-    if density == "32k" and len(data) != 91282:
+    counts = (left_cortex.index_count, right_cortex.index_count)
+    if density == "32k" and counts != (29696, 29716):
         raise ValueError("Cortex data is not in fsLR space")
 
     # medial wall needs to be added back in

--- a/niworkflows/viz/plots.py
+++ b/niworkflows/viz/plots.py
@@ -975,7 +975,7 @@ def cifti_surfaces_plot(
     surface_type="inflated",
     clip_range=(0, None),
     output_file=None,
-    **splt_kwargs,
+    **kwargs,
 ):
     """
     Plots a CIFTI-2 dense timeseries onto left/right mesh surfaces.
@@ -995,8 +995,8 @@ def cifti_surfaces_plot(
     output_file: :obj:`str` or :obj:`None`
         Path where the output figure should be saved. If this is not defined,
         then the figure will be returned.
-    splt_kwargs : dict
-        Keyword arguments for :obj:`surfplot.Plot`
+    kwargs : dict
+        Keyword arguments for :obj:`nilearn.plotting.plot_surf`
 
     Outputs
     -------
@@ -1049,13 +1049,15 @@ def cifti_surfaces_plot(
         lh_data = np.clip(lh_data, clip_range[0], clip_range[1], out=lh_data)
         rh_data = np.clip(rh_data, clip_range[0], clip_range[1], out=rh_data)
 
+    cmap = kwargs.pop('cmap', 'YlOrRd_r')
+
     # Build the figure
     lh_mesh, rh_mesh = get_surface_meshes(density, surface_type)
     figure = plt.figure(figsize=plt.figaspect(0.5))
     ax0 = figure.add_subplot(1, 2, 1, projection='3d')
-    plot_surf(lh_mesh, lh_data, cmap='YlOrRd_r', axes=ax0)
+    plot_surf(lh_mesh, lh_data, cmap=cmap, axes=ax0, **kwargs)
     ax1 = figure.add_subplot(1, 2, 2, projection='3d')
-    plot_surf(lh_mesh, lh_data, cmap='YlOrRd_r', axes=ax1)
+    plot_surf(lh_mesh, lh_data, cmap=cmap, axes=ax1, **kwargs)
 
     if output_file is not None:
         figure.savefig(output_file, bbox_inches="tight")

--- a/niworkflows/viz/plots.py
+++ b/niworkflows/viz/plots.py
@@ -1005,8 +1005,7 @@ def cifti_surfaces_plot(
     output_file: :obj:`str`
         The file where the figure is saved.
     """
-    import surfplot as splt
-    from surfplot.utils import add_fslr_medial_wall
+    from nilearn.plotting import plot_surf
 
     def get_surface_meshes(density, surface_type):
         import templateflow.api as tf
@@ -1036,28 +1035,31 @@ def cifti_surfaces_plot(
     # as potential nonsteady states
     data = img.dataobj[5:20].mean(axis=0)
 
-    cortex_data = _concat_brain_struct_data((left_cortex, right_cortex), data)
-    if density == "32k" and len(cortex_data) != 59412:
+    if density == "32k" and len(data) != 91282:
         raise ValueError("Cortex data is not in fsLR space")
-    # medial wall needs to be added back in
-    cortex_data = add_fslr_medial_wall(cortex_data)
-    if clip_range:
-        cortex_data = np.clip(cortex_data, clip_range[0], clip_range[1], out=cortex_data)
 
-    lh_data, rh_data = np.array_split(cortex_data, 2)
+    # medial wall needs to be added back in
+    lh_data = np.full(max(left_cortex.vertex_indices) + 1, np.nan)
+    rh_data = np.full(max(right_cortex.vertex_indices) + 1, np.nan)
+    lh_data[left_cortex.vertex_indices] = _concat_brain_struct_data([left_cortex], data)
+    rh_data[right_cortex.vertex_indices] = _concat_brain_struct_data([right_cortex], data)
+
+    if clip_range:
+        lh_data = np.clip(lh_data, clip_range[0], clip_range[1], out=lh_data)
+        rh_data = np.clip(rh_data, clip_range[0], clip_range[1], out=rh_data)
 
     # Build the figure
     lh_mesh, rh_mesh = get_surface_meshes(density, surface_type)
-    p = splt.Plot(
-        surf_lh=lh_mesh, surf_rh=rh_mesh, layout=splt_kwargs.pop("layout", "row"), **splt_kwargs
-    )
-    p.add_layer({'left': lh_data, 'right': rh_data}, cmap='YlOrRd_r')
-    figure = p.build()  # figsize - leave default?
+    figure = plt.figure(figsize=plt.figaspect(0.5))
+    ax0 = figure.add_subplot(1, 2, 1, projection='3d')
+    plot_surf(lh_mesh, lh_data, cmap='YlOrRd_r', axes=ax0)
+    ax1 = figure.add_subplot(1, 2, 2, projection='3d')
+    plot_surf(lh_mesh, lh_data, cmap='YlOrRd_r', axes=ax1)
 
     if output_file is not None:
         figure.savefig(output_file, bbox_inches="tight")
         plt.close(figure)
-        figure = None
+        del figure
         return output_file
 
     return figure

--- a/niworkflows/viz/plots.py
+++ b/niworkflows/viz/plots.py
@@ -1040,8 +1040,8 @@ def cifti_surfaces_plot(
         raise ValueError("Cortex data is not in fsLR space")
 
     # medial wall needs to be added back in
-    lh_data = np.full(max(left_cortex.vertex_indices) + 1, np.nan)
-    rh_data = np.full(max(right_cortex.vertex_indices) + 1, np.nan)
+    lh_data = np.full(left_cortex.surface_number_of_vertices, np.nan)
+    rh_data = np.full(right_cortex.surface_number_of_vertices, np.nan)
     lh_data[left_cortex.vertex_indices] = _concat_brain_struct_data([left_cortex], data)
     rh_data[right_cortex.vertex_indices] = _concat_brain_struct_data([right_cortex], data)
 

--- a/niworkflows/viz/plots.py
+++ b/niworkflows/viz/plots.py
@@ -1062,7 +1062,6 @@ def cifti_surfaces_plot(
     if output_file is not None:
         figure.savefig(output_file, bbox_inches="tight")
         plt.close(figure)
-        del figure
         return output_file
 
     return figure

--- a/niworkflows/viz/plots.py
+++ b/niworkflows/viz/plots.py
@@ -1058,25 +1058,40 @@ def cifti_surfaces_plot(
         mx = np.max(data)
 
     cmap = kwargs.pop('cmap', 'YlOrRd_r')
+    cbar_map = cm.ScalarMappable(norm=Normalize(mn, mx), cmap=cmap)
+
+    # Make background maps that rescale to a medium gray
+    lh_bg = np.zeros(lh_data.shape, 'int8')
+    rh_bg = np.zeros(rh_data.shape, 'int8')
+    lh_bg[:2] = [3, -2]
+    rh_bg[:2] = [3, -2]
+
+    lh_mesh, rh_mesh = get_surface_meshes(density, surface_type)
+    lh_kwargs = dict(surf_mesh=lh_mesh, surf_map=lh_data, bg_map=lh_bg)
+    rh_kwargs = dict(surf_mesh=rh_mesh, surf_map=rh_data, bg_map=rh_bg)
 
     # Build the figure
-    lh_mesh, rh_mesh = get_surface_meshes(density, surface_type)
-    figure = plt.figure(figsize=plt.figaspect(0.25))
-    ax00 = figure.add_subplot(1, 4, 1, projection='3d', rasterized=True)
-    plot_surf(lh_mesh, lh_data, hemi='left', view='lateral', cmap=cmap, axes=ax00, figure=figure, **kwargs)
-    ax00.dist = 7
-    ax01 = figure.add_subplot(1, 4, 2, projection='3d', rasterized=True)
-    plot_surf(rh_mesh, rh_data, hemi='right', view='lateral', cmap=cmap, axes=ax01, figure=figure, **kwargs)
-    ax01.dist = 7
-    ax10 = figure.add_subplot(1, 4, 3, projection='3d', rasterized=True)
-    plot_surf(lh_mesh, lh_data, hemi='left', view='medial', cmap=cmap, axes=ax10, figure=figure, **kwargs)
-    ax10.dist = 7
-    ax11 = figure.add_subplot(1, 4, 4, projection='3d', rasterized=True)
-    plot_surf(rh_mesh, rh_data, hemi='right', view='medial', cmap=cmap, axes=ax11, figure=figure, **kwargs)
-    ax11.dist = 7
+    figure = plt.figure(figsize=plt.figaspect(0.25), constrained_layout=True)
+    for i, view in enumerate(('lateral', 'medial')):
+        for j, hemi in enumerate(('left', 'right')):
+            title = f'{hemi.title()} - {view.title()}'
+            ax = figure.add_subplot(1, 4, i * 2 + j + 1, projection='3d', rasterized=True)
+            hemi_kwargs = (lh_kwargs, rh_kwargs)[j]
+            plot_surf(
+                hemi=hemi,
+                view=view,
+                title=title,
+                cmap=cmap,
+                vmin=mn,
+                vmax=mx,
+                axes=ax,
+                **hemi_kwargs,
+                **kwargs
+            )
+            # plot_surf sets this to 8, which seems a little far out, but 6 starts clipping
+            ax.dist = 7
 
-    mappable = cm.ScalarMappable(norm=Normalize(mn, mx), cmap=cmap)
-    figure.colorbar(mappable, shrink=0.2, ax=figure.axes, location='bottom')
+    figure.colorbar(cbar_map, shrink=0.2, ax=figure.axes, location='bottom')
 
     if output_file is not None:
         figure.savefig(output_file, bbox_inches="tight", dpi=400)

--- a/niworkflows/viz/plots.py
+++ b/niworkflows/viz/plots.py
@@ -1048,19 +1048,38 @@ def cifti_surfaces_plot(
     if clip_range:
         lh_data = np.clip(lh_data, clip_range[0], clip_range[1], out=lh_data)
         rh_data = np.clip(rh_data, clip_range[0], clip_range[1], out=rh_data)
+        mn, mx = clip_range
+    else:
+        mn, mx = None, None
+
+    if mn is None:
+        mn = np.min(data)
+    if mx is None:
+        mx = np.max(data)
 
     cmap = kwargs.pop('cmap', 'YlOrRd_r')
 
     # Build the figure
     lh_mesh, rh_mesh = get_surface_meshes(density, surface_type)
-    figure = plt.figure(figsize=plt.figaspect(0.5))
-    ax0 = figure.add_subplot(1, 2, 1, projection='3d')
-    plot_surf(lh_mesh, lh_data, cmap=cmap, axes=ax0, **kwargs)
-    ax1 = figure.add_subplot(1, 2, 2, projection='3d')
-    plot_surf(lh_mesh, lh_data, cmap=cmap, axes=ax1, **kwargs)
+    figure = plt.figure(figsize=plt.figaspect(0.25))
+    ax00 = figure.add_subplot(1, 4, 1, projection='3d', rasterized=True)
+    plot_surf(lh_mesh, lh_data, hemi='left', view='lateral', cmap=cmap, axes=ax00, figure=figure, **kwargs)
+    ax00.dist = 7
+    ax01 = figure.add_subplot(1, 4, 2, projection='3d', rasterized=True)
+    plot_surf(rh_mesh, rh_data, hemi='right', view='lateral', cmap=cmap, axes=ax01, figure=figure, **kwargs)
+    ax01.dist = 7
+    ax10 = figure.add_subplot(1, 4, 3, projection='3d', rasterized=True)
+    plot_surf(lh_mesh, lh_data, hemi='left', view='medial', cmap=cmap, axes=ax10, figure=figure, **kwargs)
+    ax10.dist = 7
+    ax11 = figure.add_subplot(1, 4, 4, projection='3d', rasterized=True)
+    plot_surf(rh_mesh, rh_data, hemi='right', view='medial', cmap=cmap, axes=ax11, figure=figure, **kwargs)
+    ax11.dist = 7
+
+    mappable = cm.ScalarMappable(norm=Normalize(mn, mx), cmap=cmap)
+    figure.colorbar(mappable, shrink=0.2, ax=figure.axes, location='bottom')
 
     if output_file is not None:
-        figure.savefig(output_file, bbox_inches="tight")
+        figure.savefig(output_file, bbox_inches="tight", dpi=400)
         plt.close(figure)
         return output_file
 

--- a/setup.cfg
+++ b/setup.cfg
@@ -43,7 +43,6 @@ install_requires =
     scikit-image
     scipy
     seaborn
-    surfplot
     svgutils >= 0.3.4
     transforms3d
     templateflow >= 0.7.2

